### PR TITLE
Add HLS and YouTube Live support.

### PR DESCRIPTION
**Pull request by tyler-romero**
Add support for HTTP Live Streaming (HLS) and YouTube Live videos (which is just HLS under the hood).

Also do some general cleanup:
1. remove unused imports
2. clean up motion detection documentation
3. support optional extra dependency installation
4. move from python 3.7 -> python 3.9 (oldest version that is not out of service)

### CODE REVIEW

File: src/framegrab/grabber.py [hunk 1]
```diff
@@ -1097,6 +1120,94 @@ def release(self) -> None:
...
+
+        self.type = "HLS"
+        self.config = config
+        self.hls_url = self.config["id"]["hls_url"]
+
+        self.lock = Lock()
+
+    def _apply_camera_specific_options(self, options: dict) -> None:
+        if options.get("resolution"):
+            camera_name = self.config.get("name", "Unnamed {self.type} Stream")
```
Comment (timmarkhuff) [inline src/framegrab/grabber.py:1147]: missing an f for the f string

**Comment by timmarkhuff** [comment]
I tried out the youtube live streamer and my experience is that it returns the same frame multiple times in a row, making it appear to be very low FPS. Here is what I tried:
```
import framegrab
import cv2
import time
import numpy as np

youtube_url = "https://www.youtube.com/watch?v=oJUvTVdTMyY"

config = {
    'name': 'My Youtube Grabber',
    'input_type': 'youtube_live',
    'id': {
        'youtube_url': youtube_url
    }
}

grabber = framegrab.FrameGrabber.create_grabber(config)

frame_prev = grabber.grab()
while  True:
    t1 = time.time()
    frame = grabber.grab()
    cv2.imshow('frame', frame)
    key = cv2.waitKey(1)
    t2 = time.time()
    
    elapsed_time = t2 - t1
    fps = 1 / elapsed_time
    print(f'FPS: {round(fps, 2)}')
    
    if np.array_equal(frame, frame_prev):
        print('same')
    else:
        print('different')
    frame_prev = frame.copy()
```

This is the output I saw:
```
FPS: 3.07
same
FPS: 2.53
different
FPS: 3.13
same
FPS: 3.67
same
FPS: 3.29
same
FPS: 3.72
same
FPS: 3.91
same
FPS: 3.26
same
FPS: 3.6
same
FPS: 3.24
same
FPS: 3.34
same
FPS: 3.32
same
FPS: 3.56
same
FPS: 3.14
same
FPS: 3.69
same
FPS: 4.19
same
FPS: 3.46
same
FPS: 3.13
same
FPS: 2.63
different
FPS: 3.59
same
```

**Comment by tyler-romero** [comment]
Fixed the issue! Now hitting 40 FPS with keep_connection_open=True and getting different frames

**Review by timmarkhuff — APPROVED** [review]
LGTM

**Pull request merged**